datafusion/physical-plan/src/aggregates/row_hash.rs

```
@@ -859,14 +859,13 @@ impl GroupedHashAggregateStream {
                         )?;
                     }
                     _ => {
+                        if opt_filter.is_some() {
+                            return internal_err!("aggregate filter should be applied in partial stage, there should be no filter in final stage");
```

So nice to throw rather than silently ignore something. Thank you!